sunxi: prepare H6 support and add OrangePi One Plus board

wigyori opened the pull request:
This PR moves the ATF-sunxi builds to use pre-built binaries, adds a distinction between A64/H5 and H6 for the ATF blobs used in uboot-sunxi, and adds support for the H6-based OrangePi One Plus board.

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

wigyori committed:
sunxi: add support for H6 boards and OrangePiOnePlus

Specifications:

SoC	Allwinner H6 @ 1.8 Ghz
DRAM	1GiB LPDDR3
Power	DC 5V @ 3A
Video	HDMI (Type 2.0A - full)
Audio	HDMI, on-board microphone
Network	10/100/1000Mbps Ethernet (Realtek RTL8211)
Storage	microSD
USB	1 USB2.0 Host, 1 USB2.0 OTG
Debug	Serial UART

Flashing instructions:
 Standard sunxi SD card installation procedure - copy image to SD card,
 insert in into SD card slot on the device and boot.

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

wigyori committed:
uboot-sunxi: add ATF variable for specifying which flavour to use

In preparation for H6 support (which requires a separate ATF blob), add
an envvar to the A64 boards specifying which ATF blob to use.

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

adschm commented on package/boot/arm-trusted-firmware-sunxi/Makefile:
```diff
@@ -8,19 +8,20 @@
-PKG_SOURCE_VERSION:=dbc8d9496ead9ecdd7c2a276b542a4fbbbf64027
-PKG_MIRROR_HASH:=c81521a27b86f606e927b4e346286540b862828c8d49292eae1f5c6adfc24001
+PKG_SOURCE:=atf-v$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/atf-builds/atf/releases/download/v$(PKG_VERSION)/atf-v$(PKG_VERSION).tar.gz?
```
Is the question mark at the end intended?

wigyori replied:
Yes, that was in the original patch.

adschm commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -1,3 +1,4 @@
+
```
?

adschm commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -349,7 +359,7 @@ UBOOT_TARGETS := \
 
 UBOOT_MAKE_FLAGS += \
-	BL31=$(STAGING_DIR_IMAGE)/bl31.bin
+	BL31=$(STAGING_DIR_IMAGE)/bl31_$(ATF).bin
```
Looks like we could just use UENV? Or would that not be tidy enough?

wigyori replied:
Yeah, thought about that, but I'd rather leave the two separate - there could always be a board where a separate UENV file have to be used, there were examples earlier for that.

adschm replied:
Okay, totally fine with that, we do similar stuff for build variables for devices in a few cases ...

wigyori replied:
Thanks. I'll cleanup the rest and push a new one.

adschm commented on target/linux/sunxi/image/cortexa53.mk:
```diff
@@ -14,6 +14,11 @@ define Device/sun50i-h5
   $(Device/sun50i)
 endef
 
+define Device/sun50i-h6
```
alphabetic sorting ...

adschm commented on target/linux/sunxi/image/cortexa53.mk:
```diff
@@ -70,6 +75,14 @@ define Device/pine64_pine64-plus
 endef
 TARGET_DEVICES += pine64_pine64-plus
 
+define Device/xunlong_orangepi-one-plus
```
alphabetic sorting ...

hauke commented:
I would prefer if we build the TF-A in OpenWrt when it is easy like for the sunxi platforms. 
We have https://github.com/openwrt/openwrt/blob/master/include/trusted-firmware-a.mk now in OpenWrt and we can build TF-A like it is done for mediatek: https://github.com/openwrt/openwrt/blob/master/package/boot/arm-trusted-firmware-mediatek/Makefile

hauke commented:
The FW for the power managed controller should be added as a binary as it needs a openrisc compiler.

wigyori commented:
> I would prefer if we build the TF-A in OpenWrt when it is easy like for the sunxi platforms.
> We have https://github.com/openwrt/openwrt/blob/master/include/trusted-firmware-a.mk now in OpenWrt and we can build TF-A like it is done for mediatek: https://github.com/openwrt/openwrt/blob/master/package/boot/arm-trusted-firmware-mediatek/Makefile

More than happy to do it that way. As the arm-trusted-firmware-rockchip was accepted not long ago with using atf-builds as its source, I thought that Petr's patch still has some merit.

wigyori committed:
arm-trusted-firmware-sunxi: move to use the common tf-a build code and bump to latest version

Create a -a64 package for now. Moving to the latest ATF was tested on a
SoPine A64 board.

U-Boot SPL 2020.04 (Feb 19 2021 - 11:28:19 +0000)
DRAM: 2048 MiB
Trying to boot from MMC1
NOTICE:  BL31: v2.4(release):reboot-15854-g42b39a13d2
NOTICE:  BL31: Built : 22:45:53, Feb 22 2021
NOTICE:  BL31: Detected Allwinner A64/H64/R18 SoC (1689)
NOTICE:  BL31: Found U-Boot DTB at 0x4090a28, model: SoPine with baseboard
NOTICE:  PSCI: System suspend is unavailable

U-Boot 2020.04 (Feb 19 2021 - 11:28:19 +0000) Allwinner Technology

CPU:   Allwinner A64 (SUN50I)
Model: SoPine with baseboard
DRAM:  2 GiB

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

hauke commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -262,6 +276,7 @@ define U-Boot/pine64_plus
   BUILD_DEVICES:=pine64_pine64-plus
   DEPENDS:=+PACKAGE_u-boot-pine64_plus:arm-trusted-firmware-sunxi
```
don't you have to change this dependency too?

wigyori replied:
The ATF dependencies in the second patch are adjusted to arm-trusted-firmware-sunxi-a64, when the respective package is created (when the ATF build is moved to the common code).

hauke commented on package/boot/uboot-sunxi/uEnv-h6.txt:
```diff
@@ -0,0 +1,7 @@
+part uuid mmc ${mmc_bootdev}:${mmc_rootpart} uuid
+setenv loadkernel fatload mmc \$mmc_bootdev \$kernel_addr_r uImage
+setenv loaddtb fatload mmc \$mmc_bootdev \$fdt_addr_r dtb
+setenv bootargs console=ttyS0,115200 earlyprintk root=PARTUUID=${uuid} rootwait
```
The only differences to A64 is only the removed ```earlycon=uart,mmio32,0x01c28000``` correct?

wigyori replied:
That is correct.

hauke commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -349,7 +358,7 @@ UBOOT_TARGETS := \
 
 UBOOT_MAKE_FLAGS += \
-	BL31=$(STAGING_DIR_IMAGE)/bl31.bin
+	BL31=$(STAGING_DIR_IMAGE)/bl31_$(ATF).bin
```
The ```bl31_$(ATF).bin``` is only created in a later patch, it would bet better to have this compiling all the time.

wigyori replied:
The ATF variable is created in the very first patch, and the UBOOT_MAKE_FLAGS is updated accordingly in the same one.

hauke commented:
This looks good to me in general, just some small comments.

zxlhhyccc commented:
![image](https://user-images.githubusercontent.com/45259624/114659667-99d31e00-9d26-11eb-8f94-945a10662da7.png)
